### DISCUSSION TITLE
Cannot filter on subsites / values in a list / Modern PnP Search

### DISCUSSION THREAD

**Discussion by wejden-yahyaoui** [microsoft-search/pnp-modern-search#3484] [General]
I have this list : 
![2024-01-30_16h17_09](https://github.com/microsoft-search/pnp-modern-search/assets/79808614/5bfc2b20-b3d4-4152-a049-16826186acfe)

and I want in my search page , to display results and enable filtering on the subsites mentionned in this list 
![cap2](https://github.com/microsoft-search/pnp-modern-search/assets/79808614/1b9238a9-4ac5-43a1-aee4-da3a7f5c4e86)

I have here a filter on the SiteName which display all the subsites , I should have only Program and Program22 and when I choose Program he should display all the results in this site of Program. Any help please ?

**Comment by kasperbolarsen** [comment]
Hi,
Can you share your Query Template? I would expect it to be something like Path:[tenant][Site Collection][Library]

Likewise we need to know which property ( RefianbleString?) you are using in the "Sites" Refiner, and what the mapping of the RefinableString looks like :-)

**Reply by wejden-yahyaoui** [reply]
Hello, 
The query template is : {searchTerms}  RelatedHubSites:{Hub.HubSiteId} 
and in my filter I used a refinablestring based on the SiteName

**Reply by wobba** [reply]
@wejden-yahyaoui if you have mapped ows_SiteName to your RefinableString and then re-index the data you should be able to use that as a filter.

**Reply by wejden-yahyaoui** [reply]
@wobba that's what I did , but as u can see , it returns all the names of the sites and subsites ( I want only the sites mentionned in my list)  , also for Program par exemple I should have more than 2 results because it's a subsite , it contains more items

**Reply by kasperbolarsen** [reply]
Just for info: RelatedHubSites will give you both the sites in the Hub **and** the sites in any related hubs. If you want just the sites in the Hub you can use DepartmentId instead  of RelatedHubSites.

**Reply by wejden-yahyaoui** [reply]
@kasperbolarsen okay , thank you , I see , do you have any idea how can I get only the paths mentionned in the URL column of my list ? ( without typing them manually)

**Reply by kasperbolarsen** [reply]
I doubt you will be able to combine the list and the query, but if it is important to you not to manually update the query you could do something like:
Add a PowerAutomate/LogicApp flow to the list. When something happens tag each site in the list with a property in the site property bag (make sure it is searchable, I use Add-PnPPropertyBagValue and Add-PnPIndexedProperty ). This will ensure that the site collections in your list contains a property you can use in the query :-)

**Reply by wejden-yahyaoui** [reply]
okay, l'll try it , thank you so much @kasperbolarsen :))))))
